File: backend/src/main/java/com/apb/getout/service/PacklistService.java
package com.apb.getout.service;

import java.util.List;
import java.util.UUID;

import com.apb.getout.entity.Item;
import com.apb.getout.entity.Packlist;

public interface PacklistService {

	List<Packlist> findAllPacklists();

	Packlist savePacklist(Packlist packlist);

	Packlist editPacklist(UUID uuid, Packlist packlist);

	Packlist getPacklist(UUID uuid);

	void deletePacklist(UUID id);


}

File: frontend/src/app/service/packlist.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Packlist } from '../models/packlist';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class PacklistService {

  constructor(private http:HttpClient) { }

  URL = environment.apiBaseUrl + 'packlists';

  getPacklists():Observable<Packlist[]> {
    return this.http.get<Packlist[]>(this.URL);
  }

  savePacklist(packlist: Packlist): Observable<Packlist> {
    return this.http.post<Packlist>(this.URL, packlist);
  }

  getPacklist(uuid: string): Observable<Packlist> {
    return this.http.get<Packlist>(`${this.URL}/${uuid}`);
  }

  deletePacklist(uuid: string): Observable<void> {
    return this.http.delete<void>(`${this.URL}/${uuid}`);
  } 
}
